src/site/antora/modules/ROOT/pages/manual/webapp.adoc

```diff
@@ -15,311 +15,626 @@
+Problems may arise especially if xref:manual/appenders.adoc#rollingfileappender[rolling file appender] is used.
+* Web applications have a different lifecycle from the application server.
+Additional care is required to stop Log4j Core when the application is stopped.
+See <<log4j-jakarta-web>> for more details.
 
-[.note]
-====
-To avoid problems, the Log4j shutdown hook will automatically be disabled when the log4j-web jar is included.
+The second approach is more complex and requires changes to the configuration of the application server.
+See <<sharing>> for more details.
+
+[#log4j-jakarta-web]
+== Integrating with Web Applications
+
+[NOTE]
 ====
+To avoid problems, some Log4j API and Log4j Core features are automatically disabled, when running in a Jakarta EE environment. Most notably:
 
-[id=configuration]
-=== Configuration
+- the usage of `ThreadLocal` for object pooling is disabled,
+- a web-safe implementation of
+link:../javadoc/log4j-api/org/apache/logging/log4j/spi/ThreadContextMap.html[`ThreadContextMap`] is used,
+- JMX notifications are sent synchronously,
+- the JVM shutdown hook is disabled.
 
-Log4j allows the configuration file to be specified in web.xml using the `log4jConfiguration` context parameter. Log4j will search for configuration files by:
+See xref:manual/systemproperties.adoc#log4j2.isWebapp[`log4j2.isWebapp`] for more details.
+====
 
-1. If a location is provided it will be searched for as a servlet context resource. For example, if `log4jConfiguration` contains "logging.xml" then Log4j will look for a file with that name in the root directory of the web application.
-2. If no location is defined Log4j will search for a file that starts with "log4j2" in the WEB-INF directory. If more than one file is found, and if a file that starts with "log4j2-\{name}" is present, where \{name} is the name of the web application, then it will be used. Otherwise the first file will be used.
-3. The "normal" search sequence using the classpath and file URLs will be used to locate the configuration file.
+Using a logging implementation like **Log4j Core** in a Jakarta EE application requires particular care.
+Since the lifecycle of a container or web application is independent of the lifecycle of the JVM, it's important for logging resources to be properly cleaned up (database connections closed, files closed, etc.) when the container or web application shuts down.
 
-[#Servlet-3-0]
-=== Servlet 3.0 and Newer Web Applications
+To properly synchronize the lifecycles of Log4j Core and Jakarta EE applications an additional **Log4j Web** artifact is provided.
 
-A Servlet 3.0 or newer web application is any `<web-app>` whose `version` attribute has a value of "3.0" or higher. Of course, the application must also be running in a compatible web container.
+[#log4j-jakarta-web-installation]
+=== Installation
```

Comment:
Can you add an ID to all sections, please?

Reply:
Done in 6ad209e057421baa57a79de2fe594a1f92d07b51.